packages/core/echo/echo-schema/src/types.ts

```diff
@@ -25,33 +52,22 @@ export interface Identifiable {
  */
 export type Ref<T> = T | undefined;
 
-// TODO(burdon): Name?
-export type EchoReactiveObject<T> = ReactiveObject<T> & Identifiable;
-
-export const foreignKey = (source: string, id?: string): ForeignKey => ({ source, id });
-
 /**
- * Reference to an object in a foreign database.
+ * Reactive object marker interface (does not change the shape of the object.)
+ * Accessing properties triggers signal semantics.
  */
-export type ForeignKey = {
-  /**
-   * Name of the foreign database/system.
-   * g. `github.com`.
-   */
-  source?: string;
-
-  /**
-   * Id within the foreign database.
-   */
-  id?: string;
-};
+export type ReactiveObject<T> = { [K in keyof T]: T[K] };
+
+export type EchoReactiveObject<T> = ReactiveObject<T> & Identifiable;
+
+export const foreignKey = (source: string, id: string): ForeignKey => ({ source, id });
+export const foreignKeyEquals = (a: ForeignKey, b: ForeignKey) => a.source === b.source && a.id === b.id;
 
 /**
- * Echo object metadata.
+ * Utility to split meta property from raw object.
  */
-export type ObjectMeta = {
-  /**
-   * Foreign keys.
-   */
-  keys: ForeignKey[];
+export const splitMeta = <T>(object: T & WithMeta): { object: T; meta?: ObjectMeta } => {
```

Comment:
Perhaps we just have 'create' accept @meta ?

Reply:
wdym, meta was already added to create

Reply:
Hmmm. I think I prefer it separate.